io/disk/ltp_fsstress.py

```diff
@@ -139,38 +173,166 @@ def create_fs(self, l_disk, mountpoint, fstype):
+            self.log.info("checking for raid metadata")
+            cmd = "wipefs -af %s" % self.disk
+            process.system(cmd, shell=True, ignore_status=True)
+            if self.sraid.exists():
+                return False
+            return True
+        self.log.info("checking lvm_metadata on %s" % self.raid_name)
+        cmd = 'blkid -o value -s TYPE %s' % self.raid_name
+        out = process.system_output(cmd, shell=True,
+                                    ignore_status=True).decode("utf-8")
+        if out == 'LVM2_member':
+            cmd = "wipefs -af %s" % self.raid_name
+            process.system(cmd, shell=True, ignore_status=True)
+        if wait.wait_for(is_raid_deleted, timeout=10):
+            self.log.info("software raid  %s deleted" % self.raid_name)
+        else:
+            self.err_mesg.append("failed to delete sraid %s" % self.raid_name)
+
+    def delete_lv(self):
+        """
+        checks if lv/vg exists and delete them along with its metadata
+        if exists.
+        """
+        def is_lv_deleted():
+            lv_utils.lv_remove(self.vgname, self.lvname)
+            time.sleep(5)
+            lv_utils.vg_remove(self.vgname)
+            if lv_utils.lv_check(self.vgname, self.lvname):
+                return False
+            return True
+        if wait.wait_for(is_lv_deleted, timeout=10):
+            self.log.info("lv %s deleted", self.lvname)
+        else:
+            self.err_mesg.append("failed to delete lv %s" % self.lvname)
+        # checking and deleteing if lvm_meta_data exists after lv removed
+        cmd = 'blkid -o value -s TYPE %s' % self.lv_disk
+        out = process.system_output(cmd, shell=True,
+                                    ignore_status=True).decode("utf-8")
+        if out == 'LVM2_member':
+            cmd = "wipefs -af %s" % self.lv_disk
+            process.system(cmd, shell=True, ignore_status=True)
+
     def delete_fs(self, l_disk):
-        self.part_obj.unmount()
-        delete_fs = "dd if=/dev/zero bs=512 count=512 of=%s" % l_disk
-        if process.system(delete_fs, shell=True, ignore_status=True):
-            self.fail("Failed to delete filesystem on %s" % l_disk)
+        """
+        checks for disk/dir mount, unmount if mounted and checks for
+        filesystem exitance and wipe it off after dir/disk unmount.
+        :param l_disk: disk name for which you want to check the mount status
+        :return: None
+        """
+        def is_fs_deleted():
+            cmd = "wipefs -af %s" % l_disk
+            process.system(cmd, shell=True, ignore_status=True)
+            if disk.fs_exists(l_disk):
+                return False
+            return True
+
+        def is_disk_unmounted():
+            cmd = "umount %s" % l_disk
+            cmd1 = 'umount /dev/mapper/avocado_vg-avocado_lv'
+            process.system(cmd, shell=True, ignore_status=True)
+            process.system(cmd1, shell=True, ignore_status=True)
```

Comment:
Why duplicate commands ? cant we pass disk value and a single process.system(cmd, can take care of both ?

Reply:
param initialization should be outside the function always

Reply:
will put them outside of the method.